frontend/src/features/contacts/contact/create/address/Address.tsx

```diff
@@ -0,0 +1,154 @@
+import { Button, Input, Select, SelectOption } from 'components/common/form';
+import { Stack } from 'components/common/Stack/Stack';
+import { CountryCodes } from 'constants/countryCodes';
+import { useFormikContext } from 'formik';
+import useCounter from 'hooks/useCounter';
+import { Dictionary } from 'interfaces/Dictionary';
+import { ICreatePersonForm } from 'interfaces/ICreateContact';
+import React, { useCallback, useEffect } from 'react';
+import { Col, Row } from 'react-bootstrap';
+import { MdClose } from 'react-icons/md';
+import { withNameSpace } from 'utils/formUtils';
+
+import * as Styled from '../Person/styles';
+import useAddressHelpers from './useAddressHelpers';
+
+// re-export helper hooks
+export { useAddressHelpers };
+
+export interface IAddressProps {
+  namespace?: string;
+}
+
+/**
+ * Displays addresses directly associated with this Contact Person.
+ * @param {IAddressProps} param0
+ */
+export const Address: React.FunctionComponent<IAddressProps> = ({ namespace }) => {
+  const {
+    countries,
+    provinces,
+    formLabels,
+    selectedCountryCode,
+    setSelectedCountryId,
+  } = useAddressHelpers();
+
+  const { setFieldValue } = useFormikContext<ICreatePersonForm>();
+
+  // clear associated fields (province, other country name) whenever country value is changed
+  const onCountryChanged = useCallback(
+    (e: React.ChangeEvent<HTMLSelectElement>) => {
+      setSelectedCountryId(e.target.value);
+      setFieldValue(withNameSpace(namespace, 'provinceId'), '');
+      setFieldValue(withNameSpace(namespace, 'countryOther'), '');
+    },
+    [namespace, setFieldValue, setSelectedCountryId],
+  );
+
+  // this counter determines how many address lines we render in the form; e.g. street1, street2, etc
+  const { count, increment, decrement } = useCounter({ initial: 1, min: 1, max: 3 });
```

Comment:
interesting, I take it that the need to limit the total number of elements is why this hook was created instead of using formik? something like:
values.addresses.length

it does seem strange to use external state to track the length of an array in a form, why not use the length of that array directly?

Reply:
Nope, this was done to limit the street addresses fields to show. they are not an array, they have different names in the model and in the form:

- streetAddress1
- streetAddress2
- streetAddress3

The counter controls how many we show via the "add another" link

Reply:
Tried to add a comment there but maybe is not clear

Reply:
oh, ok. Wouldn't have thought of doing it that way, but makes sense now.

Reply:
well - Initially I though of having 3 vars controlled with `useState` but the counter seems more intuitive to others - maybe not I guess